app/src/main/java/clarkson/ee408/tictactoev4/MainActivity.java

```diff
@@ -26,20 +26,87 @@ public class MainActivity extends AppCompatActivity {
     private Button[][] buttons;
     private TextView status;
     private Gson gson;
+    private boolean shouldRequestMove;
+    private SocketClient socketClient;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         this.tttGame = new TicTacToe(STARTING_PLAYER_NUMBER);
         this.gson = new GsonBuilder().serializeNulls().create();
+        socketClient = SocketClient.getInstance();
+        shouldRequestMove = false;
 
         buildGuiByCode();
+        updateTurnStatus();
 
         Handler handler = new Handler();
         GameMoveTaskRunnable runnable = new GameMoveTaskRunnable(this, handler);
         handler.post(runnable);
     }
 
+    /**
++   * Sends a request to the server to ask for a game move made by the other player.
++   */
+    public void requestMove() {
+        if (!shouldRequestMove) {
+            return; // Only request moves when it's our turn
+        }
+
+        // Create Request object with type REQUEST_MOVE
+        Request request = new Request();
+        request.setType(RequestType.REQUEST_MOVE);
+        // You might want to include game state or player info
+        request.setData(""); // Add any necessary data
+
+        // Use SocketClient to send request in networkIO thread
+        AppExecutors.getInstance().networkIO().execute(() -> {
+            try {
+                Response response = socketClient.sendRequest(request);
```

Comment:
The second parameter for sendRequest is the class to deserialize the JSON to. It should be a GamingResponse because we want the move

```suggestion
                GamingResponse response = socketClient.sendRequest(request, GamingResponse.class);
```